src/platform/silabs/wifi/SiWx/WifiInterface.cpp

```diff
@@ -894,9 +863,23 @@ void wfx_dhcp_got_ipv4(uint32_t ip)
  * @return  SL_STATUS_OK if successful,
  *          SL_STATUS_FAIL otherwise
  ***********************************************************************/
-sl_status_t wfx_power_save(rsi_power_save_profile_mode_t sl_si91x_ble_state,
-                           sl_si91x_performance_profile_t sl_si91x_wifi_state) // TODO : Figure out why the extern C is necessary
+sl_status_t wfx_power_save(rsi_power_save_profile_mode_t sl_si91x_ble_state, sl_si91x_performance_profile_t sl_si91x_wifi_state)
 {
-    return (wfx_rsi_power_save(sl_si91x_ble_state, sl_si91x_wifi_state) ? SL_STATUS_FAIL : SL_STATUS_OK);
+    int32_t error = rsi_bt_power_save_profile(sl_si91x_ble_state, 0);
```

Comment:
Is there no way to merge these int32_t and sl_status_t like it was? Also, do we really need to handle negative values here?

Reply:
Meaning `error` and `status`

Reply:
`rsi_bt_power_save_profile` returns a negative value for error while sl_status_t codes are all positives so there is no clean way to merge them.

The previous impl used signed variable for both functions which i don't think is a good idea thus the split.
The only other option I see is to use `int32` for both and "hope" the sl_status_t value falls with the range which should be the case based on the size.